File: 0443-String-Compression/0443-String-Compression.java
class Solution {
    public int compress(char[] chars) {
        if (chars.length < 2) return chars.length;
        int index = 0;
        int counter = 1;
        for(int i = 1; i < chars.length; i++) {
            if (chars[i] == chars[i - 1]) {
                counter++;
            } else {
                chars[index++] = chars[i - 1];
                if (counter != 1) 
                    for (char j : String.valueOf(counter).toCharArray()) 
                        chars[index++] = j;
                counter = 1;
            }
        }
        chars[index++] = chars[chars.length - 1];
        if (counter != 1) 
            for (char j : String.valueOf(counter).toCharArray()) 
                chars[index++] = j;
        return index;
    }
}

File: 0443-String-Compression/0443-String-Compression.go
func compress(chars []byte) int {
    if len(chars) < 2 {
		return len(chars)
	}

	index := 0
	counter := 1

	for i := 1; i < len(chars); i++ {
		if chars[i] == chars[i-1] {
			counter++
		} else {
			chars[index] = chars[i-1]
			index++

			if counter != 1 {
				counterStr := strconv.Itoa(counter)
				for _, j := range counterStr {
					chars[index] = byte(j)
					index++
				}
			}

			counter = 1
		}
	}

	chars[index] = chars[len(chars)-1]
	index++

	if counter != 1 {
		counterStr := strconv.Itoa(counter)
		for _, j := range counterStr {
			chars[index] = byte(j)
			index++
		}
	}

	return index
    
}
